### PR TITLE
Add Github Workflows to Automate Build and Run e2e tests on kind cluster

### PR DESCRIPTION
**Pull request by ajaypratap003**
Resolve: https://github.com/opendatahub-io/opendatahub-operator/issues/520

## Description

Add Github Workflows to Automate Build and Run e2e tests on kind cluster
- [ ] Create workflow that builds operator bundle from the image created in OpenShift CI by [job](https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/pull-ci-opendatahub-io-opendatahub-operator-main-opendatahub-operator-pr-image-mirror) 
- [ ] Create workflow to set `Kind` cluster and install operator bundle
- [ ] Create workflow to run e2e tests in the above created cluster.

## How Has This Been Tested?

- Set QUAY_ID, QUAY_ORG and QUAY_TOKEN of your personal Quay.io account in your fork repo 
- Open a PR on fork repo with this branch
- Make sure you should have image into Quay.io `operatorhub-operator:pr-{pr number}` ,  as image build and pushed by openshift ci tool on remote branches. On fork repo openshift ci build will not run. So push image manually or add hard code IMAGE_TAG in workflow for testing.

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .github/workflows/e2e_tests.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      - 'docs/**'
+      - '**.adoc'
+      - '**.md'
+      - 'LICENSE'
+concurrency:
+  group: ${{ github.head_ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+env:
+  QUAY_ORG: opendatahub
```
Comment (VaishnaviHire) [inline .github/workflows/e2e_tests.yaml:27]: Quay org name is `opendatahub-io`

Comment (VaishnaviHire) [reply .github/workflows/e2e_tests.yaml:27]: Yeah, my bad

**Comment by ajaypratap003** [comment]
closing this PR as open new [PR](https://github.com/opendatahub-io/opendatahub-operator/pull/678) on remote repo due to Github secrets were not resolving in fork repo.

**Pull request closed without merging**